src/containers/catalog/CatalogComponent.js

```diff
@@ -16,8 +19,26 @@ export class CatalogComponent extends React.Component {
     return (
       <div>
         <h1>Catalog</h1>
-        <EntitySetList auth={this.props.auth}/>
+        <AsyncContent {...this.props.asyncState}>
+          <EntitySetList entitySets={this.props.entitySets}/>
+        </AsyncContent>
       </div>
     );
   }
-}
\ No newline at end of file
+
+  componentDidMount() {
+    this.props.requestEntitySets();
+  }
+}
+
+function mapStateToProps(state) {
+  return state.get('catalog');
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    requestEntitySets: () => { dispatch(createEntitySetListRequest()) }
```

Comment:
I think so, but we'd have to change the `ActionFactories` naming convention. There's a bit more work to do deciding exactly how we're going to structure the async actions, so I'm throwing in a TODO and leaving it until then